compiler/rustc_infer/src/infer/relate/mod.rs

```diff
@@ -0,0 +1,12 @@
+//! This module contains the definitions of most `TypeRelation`s in the type system
+//! (except for the NLL `TypeRelating`, and some relations used for diagnostics
```

Comment:
why did you not move `nll_relate` as well? I agree that diagnostics and heuristics don't necessarily have to be moved, but I do think `nll_relate` should be.

Reply:
I'll move nll_relate too

Reply:
the comment is outdated now :grin: